### PR TITLE
Created courses table and pagination

### PR DESCRIPTION
**Pull request by dmtrth25**
Created courses table with current data from the backend.
Added pagination for these items

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/56305508/3d2566e5-9371-4d7d-9cde-275a99ac0e5e

**Resolved issue: #1394**

### CODE REVIEW

**Review by MariaAfonina — COMMENTED** [review]
Can you fix this code smell?

**Review by abalanovsky — APPROVED** [review]

File: src/containers/my-courses/my-courses-container/MyCoursesContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import CourseCard from '~/components/course-card/CourseCard'
+import { styles } from '~/containers/my-courses/my-courses-container/MyCoursesContainer.styles'
+
+import { CoursesContainerProps } from '~/types'
+
+const MyCoursesContainer: FC<CoursesContainerProps> = ({ items }) => {
+  return (
+    <Box sx={styles.wrapper}>
+      {items.map((course) => (
```
Comment (Tolik170) [inline src/containers/my-courses/my-courses-container/MyCoursesContainer.tsx:12]: move map above to the variable

File: src/containers/my-courses/my-courses-container/MyCoursesContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react'
+import { Box } from '@mui/material'
+
+import CourseCard from '~/components/course-card/CourseCard'
+import { styles } from '~/containers/my-courses/my-courses-container/MyCoursesContainer.styles'
+
+import { CoursesContainerProps } from '~/types'
+
+const MyCoursesContainer: FC<CoursesContainerProps> = ({ items }) => {
```
Comment (Tolik170) [inline src/containers/my-courses/my-courses-container/MyCoursesContainer.tsx:9]: better name it MyCorsesCardsList or smth like that

File: src/pages/my-courses/MyCourses.tsx [hunk 3]
```diff
@@ -1,19 +1,74 @@
...
       <Box sx={styles.divider}></Box>
       <AddCourseWithInput />
+      {!coursesItems.length && !coursesLoading ? (
+        <NotFoundResults
+          description={t('myCoursesPage.notFound.description')}
+        />
+      ) : (
+        <MyCoursesContainer items={coursesItems} />
+      )}
+      <AppPagination
```
Comment (Tolik170) [inline src/pages/my-courses/MyCourses.tsx:66]: I think, pagination shouldn`t render if "!coursesItems.length && !coursesLoading" is true

File: src/types/course/interfaces/course.interface.ts [hunk 4]
```diff
@@ -23,5 +24,14 @@ export interface CourseSection {
   id: number
   title: string
   description: string
-  resources: CourseResources[]
+  lessons: string[]
+  quizzes: string[]
+  attachments: string[]
+}
+
+export interface GetCoursesResponse {
```
Comment (Tolik170) [inline src/types/course/interfaces/course.interface.ts:32]: use "ItemsWithCount" interface instead and pass Course as a generic

File: src/services/course-service.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { AxiosResponse } from 'axios'
+import { URLs } from '~/constants/request'
+import { axiosClient } from '~/plugins/axiosClient'
+
+interface GetCoursesParams {
```
Comment (Tolik170) [inline src/services/course-service.ts:5]: use RequestParams interface

**Review by Tolik170 — APPROVED** [review]

**Pull request merged**